holopy/core/tests/test_utils_math.py

```diff
@@ -238,41 +239,167 @@ def test_rotation_matrix_degrees():
     assert_allclose(rotation_matrix(180., 180., 180., radians = False),
                     rotation_matrix(np.pi, np.pi, np.pi))
 
-
 #test utils
-@attr('fast')
-def test_ensure_array():
-    assert_equal(ensure_array(1.0), np.array([1.0]))
-    assert_equal(ensure_array([1.0]), np.array([1.0]))
-    assert_equal(ensure_array(np.array([1.0])), np.array([1.0]))
-    len(ensure_array(1.0))
-    len(ensure_array(np.array(1.0)))
-    len(ensure_array([1.0]))
-    len(ensure_array(False))
-    len(ensure_array(xr.DataArray([12],dims='a',coords={'a':['b']})))
-    len(ensure_array(xr.DataArray([12],dims='a',coords={'a':['b']}).sel(a=['b'])))
-    len(ensure_array(xr.DataArray(12)))
-
-
-def test_choose_pool():
-    class dummy():
-        def map():
-            return None
-    assert not isinstance(choose_pool(None), (pool.BasePool, Pool))
-    assert isinstance(choose_pool(2), MultiPool)
-    assert isinstance(choose_pool('all'), MultiPool)
-    assert isinstance(choose_pool('auto'), (pool.BasePool, Pool))
-    assert not isinstance(choose_pool(dummy), (pool.BasePool, Pool))
-
-
-@attr('fast')
-def test_ensure_listlike():
-    assert ensure_listlike(None) == []
 
+class TestEnsureArray(unittest.TestCase):
+    np_array = np.array([1])
+    xr_array = xr.DataArray([2], dims='a', coords={'a':['b']})
+
+    @attr("fast")
+    def test_None_is_unchanged(self):
+        self.assertTrue(ensure_array(None) is None)
+
+    @attr("fast")
+    def test_xarray_is_unchanged(self):
+        self.assertTrue(self.xr_array.equals(ensure_array(self.xr_array)))
+
+    @attr("fast")
+    def test_listlike(self):
+        self.assertTrue(ensure_array([1]) == self.np_array)
```

Comment:
FYI the ```unittest.TestCase``` methods have an ```assertEqual``` method, which gives a nicer error message when it fails. (It says what two things are not equal, rather than just this is False).